plugin.py

```diff
@@ -1,62 +1,69 @@
 import os
 import hashlib
 import shutil
-import threading
 import urllib.request
 import sublime
 
-from sublime_lib import ActivityIndicator
+from LSP.plugin import AbstractPlugin
+from LSP.plugin import ClientConfig
+from LSP.plugin import register_plugin  # https://github.com/sublimelsp/LSP/issues/899
+from LSP.plugin import unregister_plugin  # https://github.com/sublimelsp/LSP/issues/899
+from LSP.plugin import WorkspaceFolder
+from LSP.plugin.core.typing import List, Optional, Dict
 
-from LSP.plugin.core.handlers import LanguageHandler
-from LSP.plugin.core.settings import ClientConfig, read_client_config
 
+class Lemminx(AbstractPlugin):
 
-def is_java_installed() -> bool:
-    return shutil.which("java") is not None
+    binary = None  # type: Optional[str]
+    checksum = None  # type: Optional[str]
+    url = None  # type: Optional[str]
+    version = None  # type: Optional[str]
 
+    @classmethod
+    def name(cls) -> str:
+        return 'lemminx'
```

Comment:
Should we be dropping `lsp-` prefixes? I feel it could get a bit ambiguous for some servers from the point of the status field.

Reply:
I think it's fine to have the server name, only. When looking at the clients dict, in LSP.sublime-settings it should be clear we are talking about `lsp-`. :)

Reply:
It's a discussion for a separate issue I guess.

But I'm not sure it would be so clear for `lsp-vue` for example. You would see `vue` when having vue file open but you wouldn't know that it relates to the vue server.

Reply:
Also, one could question whether normal users should see those in the status in the first place.

I would personally miss them as those often allow me to debug things but for normal users those just take space without providing extra benefit.

Reply:
It is a question of naming LSP-... packages in general. As you started with naming them by syntax, you are right. But this package was named `LSP-lemminx` rather than `LSP-xml` to follow the language server's name which is driven by this package.

Once python-language-server is outsourced to LSP-pyls, would you then keep LSP- as well?

Reply:
Having them named after languages was a bad idea. These LSP-* are language server handlers. A language server can handle multiple languages (e.g. clangd handles c, c++, objc, objc++, and cuda). It is true that some servers "claim" to be the one-and-only server for that language, e.g. elm-language-server.